enhancements/fail-forward-upgrades.md

```diff
@@ -0,0 +1,237 @@
+3. OLM supports "failing forward" to `Operator v3`.
+
+- OLM can successfully upgrade to `Operator v3` if the `Operator v2` CSV is stuck in a FAILED state.
+- OLM can successfully upgrade to `Operator v3` if the InstallPlan is stuck in a FAILED state. 
+- Existing behavior is observed when a user has not opted into the "fail forward" feature.
+
+### Graduation Criteria
+
+### Version Skew Strategy
+
+The proposed behavior is enabled by a field present in the spec of the OperatorGroup resource. Cluster admins can choose to opt into this mode as needed. As long as the operator author provided upgrade graphs are sane, there should be no issues with turning the toggle off and on in a cluster regardless of its health.
+
+## Alternatives
+
+### Alternative 1: Providing the CSV that initiated the failed upgrade to the resolver
+
+Consider the following high-level workflow:
+1. `Operator v1` is installed
+2. `Operator v1` is upgraded to `Operator v2`, which fails to install.
+3. OLM supports "failing forward" to `Operator v3` by **providing the resolver with a set of operators installed in the namespace**. `Operator v3` is identified as a valid upgrade and is installed successfully.
+
+The workflow for catalog curators/operator authors is determined by the version of the operator sent to the resolver in Step 3. In the scenario above, the operator version could be:
+- The operator that initiated the upgrade, `Operator v1`.
+- The operator that failed to install, `Operator v2`.
+
+Instead of providing `Operator v2` to the Resolver as described in this enhancement, OLM could instead provide the resolver with `Operator v1`, the operator version that initiated the upgrade. By providing the resolver with the `Operator v1`, OLM would recalculate the list of installables using the set of operators that initiated the failed upgrade. OLM would only create a new InstallPlan once the catalog was updated to include a new upgrade path for the `Operator v1`.
+
+##### Pros
+
+- The catalog curator/operator author must be aware of the failed installation, allowing them to address the issue directly by providing a new upgrade path.
+- Involving catalog curators/operator authors helps prevent unsafe upgrades.
+- Upgrade graph maintenance relies tools/concepts embraced today.
+
+#### Cons
+
+- The catalog curator/operator author must manually address each failed installation.
+- Cluster admins cannot role forward until a new catalog is published.
+- Deviates significantly from OLM's upgrade behavior today. By the time `Operator v2` has failed, most resources have been updated to include `Operator v2` as an owner. This impacts OLMs upgrade logic in multiple places and introduces additional complexity to the solution.
+
+### Alternative 2: Address with V2 APIs
+
+The V2 APIs could provide this same toggle with fewer potential side effects, but would require more effort and a couple of releases to become available.
```

Comment:
nit: should we be more specific here and say that we attempted to do this in RukPak during a sprint and determined that the approach was difficult in its current state? mainly asking for the sake of a paper trail. we have basically everything we need here already, but a bit more specificity could be nice.

Reply:
Added a few more details.